Archipelago.HollowKnight/Goals.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UnityEngine.SceneManagement;
+using ItemChanger;
+
+namespace Archipelago.HollowKnight
+{
+    public enum Goals
+    {
+        Any = 0,
+        HollowKnight = 1,
+        SealedSiblings = 2,
+        Radiance = 3,
+        Godhome = 4,
+        MAX = Godhome
+    }
+
+    abstract class Goal {
+        private static readonly Dictionary<Goals, Goal> Lookup = new Dictionary<Goals, Goal>()
+        {
+            [Goals.Any] = new AnyGoal(),
+            [Goals.HollowKnight] = new HollowKnightGoal(),
+            [Goals.SealedSiblings] = new SealedSiblingsGoal(),
+            [Goals.Radiance] = new RadianceGoal(),
+            [Goals.Godhome] = new GodhomeGoal()
+        };
+
+        public static Goal GetGoal(Goals key)
+        {
+            Goal value;
+            if (Lookup.TryGetValue(key, out value))
+            {
+                return value;
+            }
+            return null;
+        }
+
+        public abstract string Name { get; }
+        public abstract string Description { get; }
+
+        public abstract bool VictoryCondition(string sceneName);
+        public void CheckForVictory(Scene scene)
+        {
+            Archipelago.Instance.LogDebug($"Checking for victory; goal is {this.Name}; scene {scene.name}");
+            if (VictoryCondition(scene.name))
+            {
+                Archipelago.Instance.LogDebug($"Victory detected, declaring!");
+                Archipelago.Instance.DeclareVictory();
+            }
+        }
+
+        public void Select()
+        {
+            ItemChanger.Events.AddLanguageEdit(new ItemChanger.LanguageKey("Prompts", "FOUNTAIN_PLAQUE_TOP"), FountainPlaqueTopEdit);
+            ItemChanger.Events.AddLanguageEdit(new ItemChanger.LanguageKey("Prompts", "FOUNTAIN_PLAQUE_MAIN"), FountainPlaqueNameEdit);
+            ItemChanger.Events.AddLanguageEdit(new ItemChanger.LanguageKey("Prompts", "FOUNTAIN_PLAQUE_DESC"), FountainPlaqueDescEdit);
+            ItemChanger.Events.OnSceneChange += CheckForVictory;
+        }
+
+        public void Unselect()
+        {
+            {
```

Comment:
Remove extra pair of brackets.

Reply:
Thanks, VS, for adding an extra pair of brackets.